src/library_syscall.js

```diff
@@ -718,14 +723,22 @@ var SyscallsLibrary = {
     stringToUTF8(cwd, buf, size);
     return cwdLengthInBytes;
   },
+#if MEMORY64
+  __syscall_truncate64: function(path, length) {
+#else
```

Comment:
Instead of the if-else pattern, can we use `receiveI64ParamAsI32s` here? Or `receiveI64ParamAsDouble` is actually enough as the maximum file size fits in a double.

Reply:
Hmm, yes you are right we should do that... but I think we need to convert the existing `get64` usage that we see in this file to one of those.

Reply:
I'll do that before I land this change

Reply:
Thanks for the tip.. I think this change is way nicer now.  It removes the `ifdef` from the C source too since we can rely on binaryen to do the i64 splitting!